Crash when re-creating activity while text editor is shown

I noticed a crash that I think is happening since the serialization changes (or at least the serialization changes have exposed it 😀).

cc @mzorz 

### Steps to reproduce

1. Build the alpha2 branch, enable 'Don't keep activities'
2. Start a new story, and tap the `A` button to add text
3. Switch to another app and return
4. Notice the crash

### Stacktrace

```
2020-07-22 12:30:53.714 28314-28314/com.automattic.portkey E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.automattic.portkey, PID: 28314
    java.lang.RuntimeException: Unable to destroy activity {com.automattic.portkey/com.automattic.portkey.StoryComposerActivity}: kotlin.TypeCastException: null cannot be cast to non-null type com.wordpress.stories.compose.story.StoryFrameSelectorFragment
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4941)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4970)
        at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: kotlin.TypeCastException: null cannot be cast to non-null type com.wordpress.stories.compose.story.StoryFrameSelectorFragment
        at com.wordpress.stories.compose.ComposeLoopFrameActivity.updateEditMode(ComposeLoopFrameActivity.kt:1571)
        at com.wordpress.stories.compose.ComposeLoopFrameActivity.editModeRestoreAllUIControls(ComposeLoopFrameActivity.kt:1578)
        at com.wordpress.stories.compose.ComposeLoopFrameActivity.access$editModeRestoreAllUIControls(ComposeLoopFrameActivity.kt:169)
        at com.wordpress.stories.compose.ComposeLoopFrameActivity$onCreate$2$onEditTextChangeListener$1.onDone(ComposeLoopFrameActivity.kt:349)
        at com.wordpress.stories.compose.text.TextEditorDialogFragment.onDismiss(TextEditorDialogFragment.kt:80)
        at androidx.fragment.app.DialogFragment.onDestroyView(DialogFragment.java:587)
        at com.wordpress.stories.compose.text.TextEditorDialogFragment.onDestroyView(Unknown Source:0)
        at androidx.fragment.app.Fragment.performDestroyView(Fragment.java:2908)
        at androidx.fragment.app.FragmentManager.destroyFragmentView(FragmentManager.java:1344)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1264)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
        at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
        at androidx.fragment.app.FragmentManager.dispatchDestroy(FragmentManager.java:2609)
        at androidx.fragment.app.FragmentController.dispatchDestroy(FragmentController.java:330)
        at androidx.fragment.app.FragmentActivity.onDestroy(FragmentActivity.java:365)
        at androidx.appcompat.app.AppCompatActivity.onDestroy(AppCompatActivity.java:233)
        at com.wordpress.stories.compose.ComposeLoopFrameActivity.onDestroy(ComposeLoopFrameActivity.kt:655)
        at android.app.Activity.performDestroy(Activity.java:8071)
        at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1341)
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4926)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4970) 
        at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

--- a/stories/src/main/java/com/wordpress/stories/compose/ComposeLoopFrameActivity.kt
+++ b/stories/src/main/java/com/wordpress/stories/compose/ComposeLoopFrameActivity.kt
@@ -38,7 +38,6 @@ import androidx.core.content.FileProvider
 import androidx.core.view.GestureDetectorCompat
 import androidx.core.view.ViewCompat
 import androidx.fragment.app.DialogFragment
-import androidx.fragment.app.Fragment
 import androidx.lifecycle.Lifecycle.State.DESTROYED
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider

